client/blocks/qr-code-login/index.jsx

```diff
@@ -0,0 +1,198 @@
+		token: token,
+		data: encrypted,
+	} );
+
+	setAuthState( responseData.data );
+};
+
+function TokenQRCode( { tokenData } ) {
+	if ( ! tokenData ) {
+		return <QRCodePlaceholder />;
+	}
+	const { token, encrypted } = tokenData;
+	const imageSettings = {
+		src: qrCenter,
+		height: 64,
+		width: 64,
+		excavate: false,
+	};
+	return (
+		<QRCode
+			value={ `https://apps.wordpress.com/get/?campaign=login-qr-code#qr-code-login?token=${ token }&data=${ encrypted }` }
+			size={ 352 }
+			imageSettings={ imageSettings }
+		/>
+	);
+}
+
+function QRCodePlaceholder() {
+	return (
+		<div className="qr-code-login__placeholder">
+			<span className="qr-code-login__corner-box"></span>
+			<span className="qr-code-login__corner-box"></span>
+			<span className="qr-code-login__corner-box"></span>
+		</div>
+	);
+}
+
+function QRCodeLogin() {
+	const translate = useTranslate();
+	const [ tokenData, setTokenData ] = useState( getLocalTokenData() );
+	const [ authState, setAuthState ] = useState( false );
+	const currentTimer = useRef( null );
+
+	const anonymousUserId = getTracksAnonymousUserId();
+
+	// Fetch QR code data.
+	useEffect( () => {
+		fetchQRCodeData( setTokenData, tokenData, anonymousUserId );
+	}, [ setTokenData, tokenData, anonymousUserId ] );
+
+	// Fetch the Auth Data.
+	useEffect( () => {
+		clearInterval( currentTimer.current );
+		currentTimer.current = setInterval( () => {
+			fetchAuthState( setAuthState, setTokenData, tokenData, anonymousUserId );
+		}, AUTH_PULL_INTERVAL );
+
+		return () => clearInterval( currentTimer.current );
+	}, [ setAuthState, setTokenData, tokenData, anonymousUserId, currentTimer ] );
+
+	useEffect( () => {
+		if ( ! authState ) {
+			return;
+		}
+		if ( authState.auth_url ) {
+			window.location.replace( authState.auth_url );
+			return;
+		}
+
+		return null;
+	}, [ authState ] );
+
+	const steps = [
+		translate( 'Open the {{link/}} on your phone.', {
+			components: {
+				link: (
+					<a href="https://apps.wordpress.com/get/?campaign=calypso-qrcode-apps">WordPress App</a>
+				),
+			},
+		} ),
+		translate( 'Tap the My Site Tab.' ),
+		translate( 'Tap your Profile image in the top right corner of the screen.' ),
+		translate( 'Tap the {{strong}}Scan QR Code{{/strong}} option.', {
+			components: {
+				strong: <strong />,
+			},
+		} ),
+		translate( 'Point your phone to this screen to scan the code.' ),
+	];
+
+	return (
+		<Card className="qr-code-login">
+			<div className="qr-code-login__container">
+				<TokenQRCode tokenData={ tokenData } />
+			</div>
+
+			<div className="qr-code-login__instructions">
+				<h2 className="qr-code-login__heading">{ translate( 'Use QR Code to login' ) }</h2>
+				<ol className="qr-code-login__steps">
+					{ steps.map( ( step, index ) => (
+						<li key={ 'step-' + index } className="qr-code-login__step">
+							{ step }
+						</li>
+					) ) }
+				</ol>
+				<p>
+					<a href="https://apps.wordpress.com/mobile/login-via-qr-code">Need help?</a>
```

Comment:
If the link opens in the same window, won't this break out of the current flow?

Reply:
Good catch. I update the code to always use open the links in a different window.